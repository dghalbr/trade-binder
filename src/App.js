import React, { Component } from 'react';
import { BrowserRouter as Router, Link, Route, Switch } from 'react-router-dom';
import Home from './scenes/Home/Home';
import Login from './scenes/Login/Login';
import Register from './scenes/Register/Register';
import Auth from './firebase/auth';
import NavBar from './components/NavBar/NavBar';

const fireAuth = new Auth();

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isLoggedIn: false
    };
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.register = this.register.bind(this);
  }

  login(username, password) {
    fireAuth.doSignInWithEmailAndPassword(username, password);
    this.setState({ isLoggedIn: true });
  };

  logout() {
    if (fireAuth.isLoggedIn()) {
      fireAuth.doSignOut();
    }
    this.setState({ isLoggedIn: false });
  };

  register(username, password) {
    //TODO: look into firebase callbacks/api for this and log the user in after registering 
    fireAuth.doCreateUserWithEmailAndPassword(username, password);
  };

  render() {
    return (
      <Router>
        <div>
          <NavBar isLoggedIn={this.state.isLoggedIn} logout={this.logout} />
          <Switch>
            <Route exact path="/" render={state => <Home />} />
            <Route path="/login" render={state => <Login handleLogin={this.login} />} />
            <Route path="/register" render={state => <Register handleRegister={this.register} />} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;