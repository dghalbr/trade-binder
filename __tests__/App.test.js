import React from 'react';
import ReactDOM from 'react-dom';
import App from '../src/App';
import renderer from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Component: App', () => {
  // Temporarily disabling snapshot tests since UI is constantly changing.
  // it('should match its empty snapshot', () => {
  //   const tree = renderer.create(
  //     <App />
  //   ).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
