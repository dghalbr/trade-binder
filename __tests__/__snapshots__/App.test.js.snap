// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should match its empty snapshot 1`] = `
<div>
  <div
    className="NavBar"
  >
    <div
      className="NavBar-header"
    >
      <h2>
        Trade Binder
      </h2>
      <span
        className="NavBar-nav"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </span>
      <span>
        <span
          className="NavBar-nav"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </span>
        <span
          className="NavBar-nav"
        >
          <a
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
        </span>
      </span>
    </div>
  </div>
</div>
`;
